ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -245,24 +261,109 @@ static int
 gcp_file_exists(
   WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, bool *file_exists)
 {
-    WT_UNUSED(file_system);
     WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(file_exists);
 
-    return 0;
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    int ret;
+    bool exists;
+    size_t size;
+
+    // Check if object exists in the cloud.
+    if ((ret = fs->gcp_conn->object_exists(name, exists, size)) != 0) {
+        std::cerr << "gcp_file_exists: object exists request to google cloud failed." << std::endl;
+    }
+
+    if (exists) {
+        std::cerr << "gcp_file_exists: found file in google cloud." << std::endl;
+    } else {
+        std::cerr << "gcp_file_exists: object with name " << name << " does not exist."
+                  << std::endl;
+    }
+
+    return ret;
 }
 
 static int
 gcp_file_open(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name,
   WT_FS_OPEN_FILE_TYPE file_type, uint32_t flags, WT_FILE_HANDLE **file_handle_ptr)
 {
-    WT_UNUSED(file_system);
     WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(file_type);
-    WT_UNUSED(flags);
-    WT_UNUSED(file_handle_ptr);
+
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    bool exists;
+    size_t size;
+
+    *file_handle_ptr = nullptr;
+
+    // We only support opening the file in read only mode.
+    if ((flags & WT_FS_OPEN_READONLY) == 0 || (flags & WT_FS_OPEN_CREATE) != 0) {
+        std::cerr << "gcp_file_open: read-only access required." << std::endl;
+        return EINVAL;
+    }
+
+    // Currently, only data files should be being opened; although this constraint can be relaxed in
+    // the future.
+    if (file_type != WT_FS_OPEN_FILE_TYPE_DATA && file_type != WT_FS_OPEN_FILE_TYPE_REGULAR) {
+        std::cerr << "gcp_file_open: only data file and regular types supported." << std::endl;
+        return EINVAL;
+    }
+
+    // Check if object exists in the cloud.
+    if ((fs->gcp_conn->object_exists(name, exists, size) == 0) && !exists) {
+        std::cerr << "gcp_file_open: object with name " << name << " does not exist in the bucket."
+                  << std::endl;
+        return EINVAL;
+    }
+
+    // Check if there is already an existing file handle open.
+    std::vector<gcp_file_handle *>::iterator fh_iterator =
+      std::find_if(fs->fh_list.begin(), fs->fh_list.end(),
+        [name](gcp_file_handle *fh) { return strcmp(name, fh->name.c_str()) == 0; });
```

Comment:
I would use fh->name.compare(name) instead and follow CPP conventions

Reply:
addressed here: [687beb0](https://github.com/wiredtiger/wiredtiger/pull/8777/commits/687beb0e754c4e60654dfe49bdeda9dc9bf03665)